forge-ai/src/main/java/forge/ai/ComputerUtilMana.java

```diff
@@ -666,47 +666,45 @@ private static boolean payManaCost(final ManaCostBeingPaid cost, final SpellAbil
 
         List<Mana> manaSpentToPay = test ? new ArrayList<>() : sa.getPayingMana();
         List<SpellAbility> paymentList = Lists.newArrayList();
+        final ManaPool manapool = ai.getManaPool();
+
+        // Apply the color/type conversion matrix if necessary
+        manapool.restoreColorReplacements();
+        boolean ignoreColor = false, ignoreType = false;
+        CardPlayOption mayPlay = sa.getMayPlayOption();
+        if (!effect) {
+            if (sa.isSpell() && mayPlay != null) {
+                if (mayPlay.isIgnoreManaCostType()) {
+                    ignoreType = true;
+                } else if (mayPlay.isIgnoreManaCostColor()) {
+                    ignoreColor = true;
+                }
+                mayPlay.applyManaConvert(manapool);
+            } else if (sa.isActivatedAbility() && sa.getGrantorStatic() != null && sa.getGrantorStatic().hasParam("ManaConversion")) {
+                AbilityUtils.applyManaColorConversion(manapool, sa.getGrantorStatic().getParam("ManaConversion"));
+                ignoreColor = true;
+            }
+        }
+        StaticAbilityManaConvert.manaConvert(manapool, ai, sa.getHostCard(), effect ? null : sa);
 
         if (ManaPool.payManaCostFromPool(cost, sa, ai, test, manaSpentToPay)) {
             return true;    // paid all from floating mana
         }
 
         boolean purePhyrexian = cost.containsOnlyPhyrexianMana();
-
-        boolean ignoreColor = false, ignoreType = false;
-        StaticAbility mayPlay = sa.getMayPlay();
-        if (mayPlay != null) {
-            if (mayPlay.hasParam("MayPlayIgnoreColor")) {
-                ignoreColor = true;
-            } else if (mayPlay.hasParam("MayPlayIgnoreType")) {
-                ignoreType = true;
-            }
-        } else if (sa.hasParam("ActivateIgnoreColor")) {
-            ignoreColor = true;
-        }
         boolean hasConverge = sa.getHostCard().hasConverge();
+        // TODO use matrix instead of ignoreColor, ignoreType
         ListMultimap<ManaCostShard, SpellAbility> sourcesForShards = getSourcesForShards(cost, sa, ai, test,
                 checkPlayable, hasConverge, ignoreColor, ignoreType);
```

Comment:
@tool4ever can we try to check the manaPool instead of ignoreColor/ignoreType to check what colors AI needs for what shards?

imo that would makes it more flexible

Reply:
Need to test a few things, so far it seems like that part might just be obsolete now

Reply:
ignoreColor/ignoreType is used on some places later too, feel free to try to remove it

Reply:
Done - I could not find any regressions from it :)

And this also fixes another AI payment case, in this scenario it will now cast my exiled creature:
![image](https://github.com/Card-Forge/forge/assets/8506892/15a2e6a6-2021-4ae6-9bc2-0f1bcbc81369)